terra-jupyter-base/custom/edit-mode.js

```diff
@@ -186,14 +186,17 @@ define(() => {
         }
     }
 
-    function handleMetaSuccess(res) {
+    function handleMetaSuccess(res, shouldStaggerCalls) {
         toggleMetaFailureBanner(false); // sets banner for meta status
 
         const isEditMode = res.syncMode == 'EDIT';
 
         if (isEditMode) {
             handleCheckMetaResp(res); // displays modal if theres an issue in the payload
-            getLock(res); // gets lock if in edit mode
+            //if its not the first time we are acquiring lock, wait half the time until next check meta call before calling this.
```

Comment:
This might work , but I'm having difficulty following the approach. I think it would be simpler to do the following at L109:

```
// Offset from autosave timer to mitigate Welder concurrency issues.
const initialSyncDelay = 27000;

...
checkMeta();
setTimeout(() => {
  checkMeta();
  initSyncMaintainer();
}, initialSyncDelay);
```

Reply:
The approach I took staggers the lock from checkMeta and autosave. The approach you mention staggers checkMeta + lock from autosave. I've changed it to the approach you suggest, since it seems that's what we want to be doing.